common/src/main/java/io/netty/util/internal/InternalThreadLocalMap.java

```diff
@@ -164,13 +164,17 @@ public int size() {
     }
 
     public StringBuilder stringBuilder() {
-        StringBuilder builder = stringBuilder;
-        if (builder == null || builder.capacity() > STRING_BUILDER_MAX_CAPACITY) {
```

Comment:
How about use code like this at the 170 line:
```java
if(builder.capacity() > 512){
            builder.setLength(512);
            builder.trimToSize();
        }
        builder.setLength(0);
```
with this only one time array copy 512 length.